core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -848,15 +855,83 @@ public Graph build() {
         }
 
         private void updateGraphHooks(final GraphConfig config) {
-            List<GraphHook> hooks = config.getHooks();
-            if (store.isSupported(AddNamedView.class) && !hasHook(hooks, NamedViewResolver.class)) {
-                hooks.add(0, new NamedViewResolver(config.getGraphId()));
-            }
-            if (store.isSupported(AddNamedOperation.class) && !hasHook(hooks, NamedOperationResolver.class)) {
-                config.getHooks().add(0, new NamedOperationResolver(config.getGraphId()));
-            }
+            updateNamedViewResolverHook(config);
+
+            updateNamedOperationResolverHook(config);
+
+            updateFunctionAuthoriserHook(config);
+        }
+
+        private void updateFunctionAuthoriserHook(final GraphConfig config) {
+            final List<GraphHook> hooks = config.getHooks();
             if (!hasHook(hooks, FunctionAuthoriser.class)) {
-                config.getHooks().add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+                LOGGER.info("No FunctionAuthoriser hook was supplied, adding default hook.");
+                hooks.add(new FunctionAuthoriser(FunctionAuthoriserUtil.DEFAULT_UNAUTHORISED_FUNCTIONS));
+            }
+        }
+
+        private void updateNamedViewResolverHook(final GraphConfig config) {
+            validateGetFromCacheHook(
```

Comment:
I am not sure on the naming of this, the update function calls a validate function, but the validate also updates if it is valid